Tests/SwiftDocCTests/Infrastructure/SymbolGraph/SymbolGraphLoaderTests.swift

```diff
@@ -318,23 +307,143 @@ class SymbolGraphLoaderTests: XCTestCase {
+            "target": "s:EBF",
+            "targetFallback": "A.EBF"
+        }
+        """))
+        
+        let noExtensionBlockFormatExtensionGraph = (url: tempURL.appendingPathComponent("A@NEBF.symbols.json"),
+                                                    content: makeSymbolGraphString(moduleName: "A", symbols: """
+        {
+            "kind": {
+                "identifier": "swift.func",
+                "displayName": "Function"
+            },
+            "identifier": {
+                "precise": "s:EBFfunction",
+                "interfaceLanguage": "swift"
+            },
+            "pathComponents": [
+                "EBF",
+                "function"
+            ],
+            "names": {
+                "title": "function",
+            },
+            "swiftExtension": {
+                "extendedModule": "A",
+                "typeKind": "struct"
+            },
+            "accessLevel": "public"
+        }
+        """, relationships: """
+        {
+            "kind": "memberOf",
+            "source": "s:EBFfunction",
+            "target": "s:EBF",
+            "targetFallback": "A.EBF"
+        }
+        """))
+        
+        let allGraphs = [mainGraph, emptyExtensionGraph, extensionBlockFormatExtensionGraph, noExtensionBlockFormatExtensionGraph]
+        
+        for graph in allGraphs {
+            try XCTUnwrap(graph.content.data(using: .utf8)).write(to: graph.url)
+        }
+        
+        let validUndetermined = [mainGraph, emptyExtensionGraph]
+        var loader = try makeSymbolGraphLoader(symbolGraphURLs: validUndetermined.map(\.url))
+        try loader.loadAll()
+        // by default, extension graphs should be associated with the extended graph
+        XCTAssertEqual(loader.unifiedGraphs.count, 2)
+        
+        let validEBF = [mainGraph, emptyExtensionGraph, extensionBlockFormatExtensionGraph]
+        loader = try makeSymbolGraphLoader(symbolGraphURLs: validEBF.map(\.url))
+        try loader.loadAll()
+        // found extension block symbols; extension graphs should be associated with the extending graph
+        XCTAssertEqual(loader.unifiedGraphs.count, 1)
+        
+        let validNEBF = [mainGraph, emptyExtensionGraph, noExtensionBlockFormatExtensionGraph]
+        loader = try makeSymbolGraphLoader(symbolGraphURLs: validNEBF.map(\.url))
+        try loader.loadAll()
+        // found no extension block symbols; extension graphs should be associated with the extended graph
+        XCTAssertEqual(loader.unifiedGraphs.count, 3)
+        
+        let invalid = allGraphs
+        loader = try makeSymbolGraphLoader(symbolGraphURLs: invalid.map(\.url))
+        // found non-empty extension graphs with and without extension block symbols -> should throw
+        do {
+            try loader.loadAll()
+            XCTFail("SymbolGraphLoader should throw when encountering a collection of symbol graph files, where some do and some don't use the extension block format")
```

Comment:
I don't think it would be a huge problem to allow it, however, if we start to allow mixing input formats here, that just opens the door to a lot of potential edge-case bugs. One run of the Swift compiler will always result in a consistent SGF format, so I didn't think there was a reason to allow mixed formats on the Swift-DocC side. Is there any particular reason why you'd like to allow this?

Reply:
> Is there any particular reason why you'd like to allow this?

No, I don't see any use case for mixing symbol graph files with and without extension declarations. If there are use cases we can always add this support later.

It's possible that a developer could mix the formats by accident in custom workflows so it would be nice with some minimal error message to help the developer diagnose and fix that issue.

[1]: https://github.com/apple/swift-docc/pull/369#discussion_r967365129